src/formatDate.js

```diff
@@ -51,6 +51,75 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dividerIndex = toFormat.length - 1;
+  const NEW_DIVIDER = toFormat[dividerIndex];
+
+  const result = [];
+  const dateCopy = date.split(fromFormat[dividerIndex]);
+
+  for (const datePart of toFormat) {
+    switch (datePart) {
+      case 'YY':
+        addYearShortFormat('YY', result);
+        break;
+
+      case 'YYYY':
+        addYearLongFormat('YYYY', result);
+        break;
+
+      case 'DD':
+        addDateElement('DD', result);
+        break;
+
+      case 'MM':
+        addDateElement('MM', result);
+        break;
+
+      default: break;
+    }
+  }
+
+  // add function above to format dd mm
+  function addDateElement(value, res) {
+    const index = fromFormat.indexOf(value);
+    const number = dateCopy[index];
+
+    res.push(number);
+  }
+
+  // add function to year format YY
+  function addYearShortFormat(value, res) {
+    let index = fromFormat.indexOf('YY');
+
+    if (index < 0) {
+      index = fromFormat.indexOf('YYYY');
+    }
+
+    let number = dateCopy[index];
+
+    if (number > 99) {
+      number = number % 100;
+    }
+    res.push(number);
+  }
+
+  // add function to year format YYYY
+  function addYearLongFormat(value, res) {
+    let index = fromFormat.indexOf('YY');
+
+    if (index < 0) {
+      index = fromFormat.indexOf('YYYY');
+    }
+
+    let number = dateCopy[index];
+
+    if (number <= 99) {
+      number = number >= 30 ? '19' + number : '20' + number;
```

Comment:
99, 30 , 19, 20 what do these magic numbers mean?
could you use variables for them?